airbyte-integrations/connectors/source-freshdesk/source_freshdesk/streams.py

```diff
@@ -0,0 +1,254 @@
+        if self._call_credit:
+            self._call_credit.consume(credit)
+    
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        self._consume_credit(self.call_credit)
+        yield from super().read_records(sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state)
+    
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        data = response.json()
+        if not data:
+            return []
+        for element in data:
+            yield element
+
+
+class IncrementalFreshdeskStream(FreshdeskStream, IncrementalMixin):
+
+    state_filter = "updated_since"  # Name of filter that corresponds to the state
+
+    def __init__(self, authenticator: AuthBase, config: Mapping[str, Any], *args, **kwargs):
+        super().__init__(authenticator=authenticator, config=config, *args, **kwargs)
+        self._state = None
+
+    @property
+    def cursor_field(self) -> str:
+        return "updated_at"
+    
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        return {"updated_at": self._state} if self._state else {}
+
+    @state.setter
+    def state(self, value: MutableMapping[str, Any]):
+        if "updated_at" in value:
+            self._state = value["updated_at"]
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        if "updated_at" in stream_state:
+            params[self.state_filter] = stream_state["updated_at"]
+        else:
+            params[self.state_filter] = self.start_date
+        return params
+    
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state):
+            yield record
+            if not self.state or self.state["updated_at"] < record[self.cursor_field]:
+                self.state = record
+
+
+class Agents(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "agents"
+
+
+class Companies(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "companies"
+
+
+class Contacts(IncrementalFreshdeskStream):
+    state_filter = "_updated_since"
+    state_checkpoint_interval = 100
```

Comment:
I think you can set this checkpoint interval at the `IncrementalFreshdeskStream` level

Reply:
Sounds good. Done.